src/main/java/org/example/CalendarManagement/calendarfacade/MeetingFacade.java

```diff
@@ -0,0 +1,121 @@
+    @Autowired
+    MeetingServiceClient meetingServiceClient;
+
+    @Autowired
+    MeetingRoomRepository meetingRoomRepository;
+
+    @Autowired
+    EmployeeRepository employeeRepository;
+
+    @Transactional
+    public Response scheduleMeeting(AddMeetingDataRequest addMeetingDataRequest){
+        Response response =null;
+        Date dateOfMeeting = new Date(addMeetingDataRequest.getDateOfMeeting().getDayOfMonth(),addMeetingDataRequest.getDateOfMeeting().getMonthValue(),addMeetingDataRequest.getDateOfMeeting().getYear());
+        Time meetingStartTime = new Time(addMeetingDataRequest.getStartTime().getHour(),addMeetingDataRequest.getStartTime().getMinute(),addMeetingDataRequest.getStartTime().getSecond());
+        Time meetingEndTime = new Time(addMeetingDataRequest.getEndTime().getHour(),addMeetingDataRequest.getEndTime().getMinute(),addMeetingDataRequest.getEndTime().getSecond());
+        EmployeeAvailabilityDataRequest employeeAvailabilityDataRequest = new EmployeeAvailabilityDataRequest(
+                addMeetingDataRequest.getListOfEmployeeId(),
+                meetingStartTime,
+                meetingEndTime,
+                dateOfMeeting
+        );
+        List<String> employeesNotAvailable = meetingServiceClient.checkEmployeeAvailability(employeeAvailabilityDataRequest);
+        if(employeesNotAvailable.size()>0){
+            response = new Response("these employees are not available", employeesNotAvailable);
+            return response;
+        }
+
+        Integer meetingRoomId = 0;
+        if(addMeetingDataRequest.getRoomName().equals(""))
+        {
+            int officeId = employeeRepository.findOfficeIdById(addMeetingDataRequest.getOwnerId());
+            meetingRoomId =  meetingRoomService.findFreeMeetingRoom(officeId,addMeetingDataRequest.getDateOfMeeting(),addMeetingDataRequest.getStartTime(),addMeetingDataRequest.getEndTime());
+
+            if(meetingRoomId == 0)
+            {
+                response = new Response("No Meeting room is available", "");
+                return response;
+            }
+            response = new Response(null, meetingRoomId);
+        }
+        else
+        {
+            boolean meetingRoomAvailableResponse = meetingRoomService.meetingRoomAvailable(addMeetingDataRequest.getRoomName(),
+                    addMeetingDataRequest.getDateOfMeeting(),addMeetingDataRequest.getStartTime(),addMeetingDataRequest.getEndTime());
+
+            if(!meetingRoomAvailableResponse)
+            {
+                response = new Response("Given Meeting room is not available","");
+                return response;
+            }
+            Optional<MeetingRoom> meetingRoom = meetingRoomRepository.findByName(addMeetingDataRequest.getRoomName());
+            meetingRoomId = meetingRoom.get().getRoomId();
+            response = new Response(null, meetingRoom.get().getRoomName());
+        }
+
+        MeetingDetails meetingDetails = new MeetingDetails();
```

Comment:
pass the request object to thrift service, thrift service can call thrfit object mapper service which can map this request to MeetingDetails object

Reply:
Done